docs/openfaas-pro/iam/function-authentication.md

```diff
@@ -0,0 +1,275 @@
+            token_url = "https://gateway.example.com/oauth/token"
+
+            # Create URI object
+            uri = URI(token_url)
+
+            # Create the HTTP object
+            http = Net::HTTP.new(uri.host, uri.port)
+            http.use_ssl = true if uri.scheme == 'https'
+
+            # Define the request parameters
+            request_params = {
+                'subject_token' => token,
+                'subject_token_type' => 'urn:ietf:params:oauth:token-type:id_token',
+                'grant_type' => 'urn:ietf:params:oauth:grant-type:token-exchange',
+                'scope' => 'functions'
+            }
+
+            # Create the POST request
+            request = Net::HTTP::Post.new(uri.request_uri)
+            request.set_form_data(request_params)
+            request['Content-Type'] = 'application/x-www-form-urlencoded'
+
+            # Execute the request
+            response = http.request(request)
+
+            # Parse the response
+            result = JSON.parse(response.body)
+
+            # Check the response for errors
+            if response.code != '200'
+                puts "Error: #{response.code}"
+                raise "Failed to retrieve token: #{result['error_description'] || 'Unknown error'}"
+            end
+
+            # Return the access token
+            result['access_token']
+        end
+        ```
+
+3. Invoke an authenticated function
+
+    Invoke an authenticated function with the function access token.
+
+    === "curl"
+
+        ```bash
+        curl -i "https://gateway.example.com/function/env" \
+            -H "Authorization: Bearer $(cat ./function-token.txt)"
+        ```
+
+    === "ruby"
+
+        ```ruby
+        def invoke_function(token)
+            function_url = "https://gateway.example.com/function/env"
+
+            # Create URI object
+            uri = URI(function_url)
+
+            # Create the HTTP object
+            http = Net::HTTP.new(uri.host, uri.port)
+            http.use_ssl = true if uri.scheme == 'https'
+
+            # Create the GET request
+            request = Net::HTTP::Get.new(uri.request_uri)
+            request['Authorization'] = "Bearer: #{token}"
+
+            # Execute the request
+            response = http.request(request)
+
+            # Check the response for errors
+            if response.code != '200'
+                puts "Error: #{response.code}"
+                raise "Failed to invoke function: code #{response.code}, body: #{response.body}"
+            end
+
+            # Return response body
+            response.body
+        end
+        ```
+
+### Invoke authenticated functions with the faas-cli
+ 
+The `faas-cli invoke` command can be used to invoke authenticated functions:
+
+```bash
+echo "OpenFaaS" | faas-cli invoke figlet
+```
+
+You will only be able to invoke authenticated function if the CLI is authenticated with the gateway, see: [SSO with CLI](/openfaas-pro/sso/cli/)
```

Comment:
Also mention `faas-cli pro auth` command so there's no need to jump off and read a new page.